### PR TITLE
feature control music

### PR DESCRIPTION
**Pull request by VanHieu7119**
feature control music: loop, loop 1, noon loop, shuffle, non shuffle, auto play
Link Redmine: https://edu-redmine.framgia.vn/issues/2703

### CODE REVIEW

**Comment by VanHieu7119** [comment]
ready

File: app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java [hunk 1]
```diff
@@ -254,34 +262,56 @@ public void setButtonPlay() {
...
+    public void handlePrevious(){
+        if (mState == ONE) {
+            mServicePlayMusic.playMusic();
+        } else {
+            if (mPosition == ZERO) {
+                mPosition = mSongList.size() - ONE;
+            } else {
+                mPosition--;
+            }
+            if (mIsShuffle == ONE) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java]: refactor code 2 method handleNext & handlePrevious lại e

File: app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java [hunk 2]
```diff
@@ -295,16 +325,44 @@ public void setButtonPlayMini() {
...
+                break;
+            case ONE:
+                mState = TWO;
+                mButtonLoop.setImageResource(R.drawable.ic_repeat_24dp);
+                Toast.makeText(getContext(), "Loop All", Toast.LENGTH_SHORT).show();
+                break;
+            case TWO:
+                mState = ZERO;
+                mButtonLoop.setImageResource(R.drawable.ic_non_repeat_24dp);
+                Toast.makeText(getContext(), "Non Loop", Toast.LENGTH_SHORT).show();
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java]: as above

File: app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java [hunk 3]
```diff
@@ -295,16 +325,44 @@ public void setButtonPlayMini() {
...
+        switch (mState) {
+            case ZERO:
+                mState = ONE;
+                mButtonLoop.setImageResource(R.drawable.ic_repeat_one_24dp);
+                Toast.makeText(getContext(), "Loop One", Toast.LENGTH_SHORT).show();
+                break;
+            case ONE:
+                mState = TWO;
+                mButtonLoop.setImageResource(R.drawable.ic_repeat_24dp);
+                Toast.makeText(getContext(), "Loop All", Toast.LENGTH_SHORT).show();
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java]: as above

File: app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java [hunk 4]
```diff
@@ -295,16 +325,44 @@ public void setButtonPlayMini() {
...
         setUISong();
     }
 
+    private void setButtonLoop() {
+
+        switch (mState) {
+            case ZERO:
+                mState = ONE;
+                mButtonLoop.setImageResource(R.drawable.ic_repeat_one_24dp);
+                Toast.makeText(getContext(), "Loop One", Toast.LENGTH_SHORT).show();
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java]: hard code. you should remove Toast

File: app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java [hunk 5]
```diff
@@ -60,6 +66,9 @@
     private boolean mIsBoundService;
     private View mView;
     private View mViewListMusic;
+    private int mState = ZERO;
+    private int mIsShuffle = ZERO;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java]: để = 0 như bình thường e. tương tự biến mState

File: app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java [hunk 6]
```diff
@@ -295,16 +325,44 @@ public void setButtonPlayMini() {
...
+                Toast.makeText(getContext(), "Non Loop", Toast.LENGTH_SHORT).show();
+                break;
+        }
+    }
+
+    private void setButtonShuffle() {
+        if (mIsShuffle == ZERO) {
+            mIsShuffle = ONE;
+            mButtonShuffle.setImageResource(R.drawable.ic_shuffle_black_24dp);
+            Toast.makeText(getContext(), "Shuffle", Toast.LENGTH_SHORT).show();
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java]: as above

File: app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java [hunk 7]
```diff
@@ -295,16 +325,44 @@ public void setButtonPlayMini() {
...
+        } else {
+            mIsShuffle = ZERO;
+            mButtonShuffle.setImageResource(R.drawable.ic_non_shuffle_24dp);
+            Toast.makeText(getContext(), "Non Shuffle", Toast.LENGTH_SHORT).show();
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java]: as above

**Comment by tuanlvt-1117** [comment]
commented

**Comment by VanHieu7119** [comment]
ready

File: app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java [hunk 8]
```diff
@@ -329,14 +388,40 @@ public void onClick(View v) {
...
     }
 
     @Override
     public void onCompletion(MediaPlayer mp) {
-        mButtonPlayMini.setImageResource(R.drawable.ic_play_arrow_white_24dp);
-        mButtonPlay.setImageResource(R.drawable.ic_play_arrow_black_24dp);
-        mRotateAnimation.setRepeatCount(0);
-        mImageMusic.clearAnimation();
+        switch (mState) {
+            case 0:
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_39/screen/playmusic/FragmentPlayMusic.java]: tạo @IntDef cho các case loop, shuffle đi e

**Comment by tuanlvt-1117** [comment]
commented

**Comment by VanHieu7119** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Comment by tuanlvt-1117** [comment]
merged

**Pull request merged**